VitruvioHost/Plugins/Vitruvio/Source/VitruvioEditor/Private/VitruvioComponentDetails.cpp

```diff
@@ -760,30 +892,46 @@ void FVitruvioComponentDetails::BuildAttributeEditor(IDetailLayoutBuilder& Detai
 		TArray<UObject*> Objects;
 		Objects.Add(Attribute);
 		Generator->SetObjects(Objects);
-		Generator->OnFinishedChangingProperties().AddLambda([this, VitruvioActor, Attribute](FPropertyChangedEvent Event) {
-			if (Event.ChangeType == EPropertyChangeType::ArrayAdd)
-			{
-				if (UArrayAttribute* ArrayAttribute = Cast<UArrayAttribute>(Attribute))
+		Generator->OnFinishedChangingProperties().AddLambda([this, VitruvioActor, Attribute, AttributeKey](FPropertyChangedEvent Event) {
+				if (Event.ChangeType == EPropertyChangeType::ArrayAdd)
 				{
-					Event.ObjectIteratorIndex = 0;
-					const int ArrayIndex = Event.GetArrayIndex(Event.Property->GetFName().ToString());
-					ArrayAttribute->InitializeDefaultArrayValue(ArrayIndex);
+					if (UArrayAttribute* ArrayAttribute = Cast<UArrayAttribute>(Attribute))
+					{
+						Event.ObjectIteratorIndex = 0;
+						const int ArrayIndex = Event.GetArrayIndex(Event.Property->GetFName().ToString());
+						ArrayAttribute->InitializeDefaultArrayValue(ArrayIndex);
+					}
 				}
-			}
-			Attribute->bUserSet = true;
-			VitruvioActor->EvaluateRuleAttributes(VitruvioActor->GenerateAutomatically);
-		});
+				Attribute->bUserSet = true;
+				VitruvioActor->EvaluateRuleAttributes(VitruvioActor->GenerateAutomatically);
+
+				// Apply attribute changes to all selected vitruvio actors (if more than 1)
+				// Issue: Undo doesn't undo the change in attributes on all vitruvio actors
+				TMap<FString, FString> ChangedAttributes;
+				ChangedAttributes.Add(AttributeKey, Attribute->GetValueAsString());
+				for (size_t i = 0; i < VitruvioComponentsSelected.Num(); ++i)
+				{
+					VitruvioComponentsSelected[i]->SetAttributes(ChangedAttributes);
+					VitruvioComponentsSelected[i]->EvaluateRuleAttributes(VitruvioComponentsSelected[i]->GenerateAutomatically);
```

Comment:
It could make sense to introduce a generic `SetAttribute` function on the `VitruvioComponent` (which takes a two Strings key and value) (internally the SetAttributes could then use this SetAttribute function to reduce code duplication).

Then we wouldn't have to abuse the `SetAttributes` function?

Not necessary to change this. What do you think?